src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -266,4 +284,28 @@ public boolean exists() throws RealmUnavailableException {
             return false;
         }
     };
+
+    /**
+     * Represents a key that is specific to the realm and used to uniquely identify a {@link RealmIdentity} instance.
+     */
+    interface Key {
+
+        Key EMPTY = (Key) () -> null;
+
+        /**
+         * Returns the value of this key.
+         *
+         * @return the value of this key
+         */
+        Object getValue();
```

Comment:
Humm ... Good catch. Indeed, it is cleaner and avoid dealing with different "key locators" (name + id).

I'm just not sure about dropping IdentityLocator, it might be useful in case we want to support future requirements around identity lookup. What about you ?